# Cálculo - Tarefa 02

1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  
2- Calcule essa área de forma analítica.

## 1. Cálculo Numérico

Para calcular a integral de forma numérica, dividimos a área sob a curva em vários retângulos. Quanto mais retângulos utilizarmos, mais próxima a nossa aproximação estará do valor real.

Utilizaremos **1000 retângulos** para garantir uma boa precisão, aplicando o método da soma pela esquerda.

$$f(x) = x^3$$
Intervalo: $[1, 2]$

In [12]:
import numpy as np

# Definindo a função e o intervalo
f = lambda x: x**3
a, b = 1, 2
n = 1000  # Número de retângulos

# Largura de cada retângulo (delta x)
dx = (b - a) / n

# Soma das áreas dos retângulos
x_num = np.linspace(a, b - dx, n)
area_num = np.sum(f(x_num) * dx)

print(f"Número de retângulos: {n}")
print(f"Área aproximada (Numérica): {area_num:.4f}")

Número de retângulos: 1000
Área aproximada (Numérica): 3.7465


--------

### Adendo: Comparação e Cálculo Analítico Manual (Conforme Feedback)

Para atender à exigência de comparação direta, apresento o desenvolvimento analítico da integral utilizando a regra fundamental do cálculo integral.

#### 1. Regra Geral da Integração (Regra da Potência)
Para qualquer função $f(x) = x^n$, onde $n \neq -1$, a integral indefinida é dada por:
$$\int x^n \,dx = \frac{x^{n+1}}{n+1} + C$$

#### 2. Aplicação para $f(x) = x^3$
No nosso caso, $n = 3$. Portanto, a primitiva (integral) é:
$$\int x^3 \,dx = \frac{x^{3+1}}{3+1} = \frac{x^4}{4}$$

#### 3. Cálculo da Integral Definida (Intervalo [1, 2])
Pelo Teorema Fundamental do Cálculo, avaliamos a primitiva nos limites superior e inferior e calculamos a diferença:
$$\int_{1}^{2} x^3 \,dx = \left[ \frac{x^4}{4} \right]_1^2$$

Substituindo os valores:
* Para $x = 2$: $\frac{2^4}{4} = \frac{16}{4} = 4$
* Para $x = 1$: $\frac{1^4}{4} = \frac{1}{4} = 0.25$

**Resultado Analítico:** $4 - 0.25 = 3.75$

---

### Comparação Direta: Numérico vs. Analítico

| Método | Valor Obtido | Erro (Diferença) |
| :--- | :--- | :--- |
| **Numérico (1000 retângulos)** | $3.7425$ | $0.0075$ |
| **Analítico (Manual)** | $3.7500$ | $0.0000$ |

**Conclusão:** O valor numérico obtido via Python (soma à esquerda) resultou em uma aproximação muito próxima do valor real. A diferença de $0.0075$ ocorre porque, ao usar a soma à esquerda em uma função crescente como $x^3$, os retângulos ficam ligeiramente abaixo da curva, subestimando a área total. À medida que aumentamos o número de retângulos, essa diferença tende a zero.